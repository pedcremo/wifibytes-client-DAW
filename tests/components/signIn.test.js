import SignIn from '../../src/components/login/signIn';
import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

beforeEach(() => {
    // Set up our document body
    document.body.innerHTML =
        `<div id="main" class="container-fluid pl-0 pr-0">
        </div>`;
});

it('SignIn component renders the signIn component correctly', () => {
    const rendered = Enzyme.shallow(<SignIn />);
    expect(rendered).toHaveLength(1);
});

it('We can check if SignIn component called the class constructor', () => {
    const signIn = Enzyme.shallow(<SignIn />);
    expect(signIn).toMatchSnapshot();
});

it("When click Change Set Render", () => {
  const signIn = Enzyme.shallow(<SignIn />);
  signIn.setState({
      url : "login"
  });
  expect(signIn.state('url')).toBe("login");
  signIn.find('#register').simulate('click', signIn.setState({url : "register"}));
  expect(signIn.state('url')).toBe("register");
});
