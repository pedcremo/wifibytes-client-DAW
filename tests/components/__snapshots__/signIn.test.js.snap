// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`We can check if SignIn component called the class constructor 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignIn />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="center"
        >
          <div
            className="tab"
          >
            <button
              className="tablinks"
              id="loginButton"
              onClick={[Function]}
            >
              Login
            </button>
            <button
              className="tablinks"
              id="registerButton"
              onClick={[Function]}
            >
              Register
            </button>
          </div>
        </div>,
        <Login />,
      ],
      "className": "login",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="tab"
          >
            <button
              className="tablinks"
              id="loginButton"
              onClick={[Function]}
            >
              Login
            </button>
            <button
              className="tablinks"
              id="registerButton"
              onClick={[Function]}
            >
              Register
            </button>
          </div>,
          "className": "center",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="tablinks"
                id="loginButton"
                onClick={[Function]}
              >
                Login
              </button>,
              <button
                className="tablinks"
                id="registerButton"
                onClick={[Function]}
              >
                Register
              </button>,
            ],
            "className": "tab",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Login",
                "className": "tablinks",
                "id": "loginButton",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Login",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Register",
                "className": "tablinks",
                "id": "registerButton",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Register",
              "type": "button",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="center"
          >
            <div
              className="tab"
            >
              <button
                className="tablinks"
                id="loginButton"
                onClick={[Function]}
              >
                Login
              </button>
              <button
                className="tablinks"
                id="registerButton"
                onClick={[Function]}
              >
                Register
              </button>
            </div>
          </div>,
          <Login />,
        ],
        "className": "login",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="tab"
            >
              <button
                className="tablinks"
                id="loginButton"
                onClick={[Function]}
              >
                Login
              </button>
              <button
                className="tablinks"
                id="registerButton"
                onClick={[Function]}
              >
                Register
              </button>
            </div>,
            "className": "center",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <button
                  className="tablinks"
                  id="loginButton"
                  onClick={[Function]}
                >
                  Login
                </button>,
                <button
                  className="tablinks"
                  id="registerButton"
                  onClick={[Function]}
                >
                  Register
                </button>,
              ],
              "className": "tab",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Login",
                  "className": "tablinks",
                  "id": "loginButton",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Login",
                "type": "button",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Register",
                  "className": "tablinks",
                  "id": "registerButton",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": "Register",
                "type": "button",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
